torchvision/prototype/transforms/_utils.py

```diff
@@ -80,7 +97,7 @@ def _check_padding_mode_arg(padding_mode: Literal["constant", "edge", "reflect",
 
 
 def query_bounding_box(flat_inputs: List[Any]) -> features.BoundingBox:
-    bounding_boxes = {inpt for inpt in flat_inputs if isinstance(inpt, features.BoundingBox)}
+    bounding_boxes = [inpt for inpt in flat_inputs if isinstance(inpt, features.BoundingBox)]
```

Comment:
I think we can use a list here instead of a set.

Reply:
Is this a perf optimization? Otherwise I would prefer the set since

1. it makes it more clear we are looking for duplicates
2. it is aligned with the other `query_*` function below.

Functionally, the only difference is that with a `set`, passing the same bounding box twice in the same sample would not be caught. This indeed seems like a user error, but if we raise there, we should probably check for duplicates everywhere. Not sure if we want to single out bounding boxes here.

Reply:
It's both perf and code correctness. What we want here is to ensure there is only 1 bbox in the input. If there were two, even if they were the same exact copy, our transforms that rely on `query_bounding_box` would end up working incorrectly. Effectively they would end up modifying the first copy but not the second. This is a different use-case from the rest of the `query_*` methods where we are able to handle multiple entries but just want to ensure same size across.

Reply:
As discussed offline, `query_chw` and `query_spatial_size` must use sets because what we are interested in is enforcing that all images have a single size. Multiple images/videos are allowed in these cases, we just enforce the size is the same everywhere.

On the other hand the `query_bounding_box` is used to extract the single bbox. The transforms that use the specific util, handle only 1 bbox and this is why we need a list. Note that this is identical to the `_flatten_and_extract_image_or_video` from AA where we also use a list.

So this is primarily a bug fix to restore the right semantics on the function and also has secondary positive performance implications.